interfaces/hotplug/udevadm.go

```diff
@@ -0,0 +1,110 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package hotplug
+
+import (
+	"bufio"
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strings"
+)
+
+var udevadmBin = `/sbin/udevadm`
+var udevadmRe = regexp.MustCompile("^([A-Z]): (.*)$")
+
+// RunUdevadm enumerates all devices by parsing 'udevadm info -e' command output and reports them
+// via devices channel. The devices channel gets closed to indicate that all devices were processed.
+// Parsing errors are reported to parseErrors channel. Fatal errors encountered when starting udevadm
+// are reported by the error return value.
+func RunUdevadm(devices chan<- *HotplugDeviceInfo, parseErrors chan<- error) error {
+	cmd := exec.Command(udevadmBin, "info", "-e")
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		close(devices)
+		return err
+	}
+
+	rd := bufio.NewScanner(stdout)
+	if err := cmd.Start(); err != nil {
+		close(devices)
+		return err
+	}
+
+	go func() {
+		defer close(devices)
+
+		env := make(map[string]string)
+		for rd.Scan() {
```

Comment:
A comment like this would be useful:
```
// udevadm export output is divided in per-device blocks, blocks are separated
// with empty lines, each block starts with device path (P:) line, within a block 
// there is one attribute per line, example:
//
// P: /devices/virtual/workqueue/nvme-wq
// E: DEVPATH=/devices/virtual/workqueue/nvme-wq
// E: SUBSYSTEM=workqueue
// <empty-line>
// P: /devices/virtual/block/dm-1
// N: dm-1
// S: disk/by-id/dm-name-linux-root
// E: DEVNAME=/dev/dm-1
// E: USEC_INITIALIZED=8899394
// <empty-line>
```

Reply:
Added